Modules/pyexpat.c

```diff
@@ -92,22 +92,34 @@ typedef struct {
     PyObject **handlers;
 } xmlparseobject;
 
+#define xmlparseobject_CAST(op) ((xmlparseobject *)(op))
+
 #include "clinic/pyexpat.c.h"
 
 #define CHARACTER_DATA_BUFFER_SIZE 8192
 
-typedef void (*xmlhandlersetter)(XML_Parser self, void *meth);
-typedef void* xmlhandler;
+typedef const void* xmlhandler;
+typedef void (*xmlhandlersetter)(XML_Parser self, xmlhandler handler);
 
 struct HandlerInfo {
     const char *name;
-    xmlhandlersetter setter;
+    // To avoid UBSan failures, we cannot use the 'xmlhandlersetter' type,
+    // as the signature of the setters will not be compatible. However, we
+    // can safely convert the pointers since a setter function will be passed
+    // a correct handler.
+    const void *setter;
```

Comment:
This relies on casting between *data* and *function* pointers, which has undefined behaviour. CPython does that a lot of course, but in a PR that removes UB, it would be more appropriate to use a function pointer (`xmlhandlersetter`) here.

Reply:
> This relies on casting between data and function pointers

Oh is it also UB ? :( 

Ok I'll revert this and keep a comment.